src/d3a/d3a_core/sim_results/kpi.py

```diff
@@ -119,30 +121,32 @@ def _dissipate_infinite_consumption(self, trade):
         demanded_buffer_wh also needs to accumulated accounting of what
         the InfiniteBus has consumed/demanded.
         """
-        if trade.buyer_origin in self.buffer_list and trade.seller_origin in self.producer_list:
-            self.total_self_consumption_wh += trade.offer.energy * 1000
-            self.demanded_buffer_wh += trade.offer.energy * 1000
-
-    def _accumulate_energy_trace(self):
-        for c_area in self.areas_to_trace_list:
-            if c_area not in self.accounted_markets:
-                self.accounted_markets[c_area] = []
-            for market in c_area.past_markets:
-                if market.time_slot_str in self.accounted_markets[c_area]:
-                    continue
-                self.accounted_markets[c_area].append(market.time_slot_str)
-                for trade in market.trades:
-                    self._accumulate_self_consumption(trade)
-                    self._accumulate_self_production(trade)
-                    self._accumulate_self_consumption_buffer(trade)
-                    self._dissipate_self_consumption_buffer(trade)
-                    self._accumulate_infinite_consumption(trade)
-                    self._dissipate_infinite_consumption(trade)
-
-    def update_area_kpi(self, area):
+        if trade['buyer_origin'] in self.buffer_list and \
+                trade['seller_origin'] in self.producer_list:
+            self.total_self_consumption_wh += trade['energy'] * 1000
+            self.demanded_buffer_wh += trade['energy'] * 1000
+
+    def _accumulate_energy_trace(self, core_stats):
+        for target_area_uuid in self.areas_to_trace_list:
+            target_core_stats = core_stats.get(target_area_uuid, {})
+            # if c_area not in self.accounted_markets:
+            #     self.accounted_markets[c_area] = []
+            # for market in c_area.past_markets:
+            #     if market.time_slot_str in self.accounted_markets[c_area]:
+            #         continue
+            #     self.accounted_markets[c_area].append(market.time_slot_str)
```

Comment:
Can this be removed?

Reply:
I guess not. It was added in the frame of https://gridsingularity.atlassian.net/browse/D3ASIM-2103

Reply:
Already covered by three [integrationtests](https://github.com/gridsingularity/d3a/blob/master/integration_tests/kpi.feature#L45) targeting 2103; And since now results are only calculated for current market, populating `accounted_markets ` for past_markets could safely be skipped.

Reply:
ok, please also remove the unneeded member `accounted_markets` member of the class then. Thanks